### ISSUE TITLE
BUG when using force flag for event deployment

### ISSUE THREAD

**Issue by maikia** [paris-saclay-cds/ramp-board#428]
when deploying a new event with a --force flag:
`ramp setup deploy-event --event-config events/iris_test/config.yml --no-cloning --force`

the event is being closed properly to deploy it again (here iris_test), however the other events which are using the same starting kit (here all iris events) are also closed

**Comment by lucyleeow** [comment]
Hmm it says [here](https://paris-saclay-cds.github.io/ramp-docs/ramp-board/stable/cli.html?highlight=force#cmdoption-ramp-database-add-event-force) that the `--force` flag:

> Whether or not to overwrite the problem if it exists [default: False]

So maybe we actually don't need to add that to re-deploy?

**Comment by maikia** [comment]
But if the things changed and if we need to overwrite it should not kill all the events, just the one we want to change

**Comment by lucyleeow** [comment]
Ah I see, I misunderstood!

**Comment by maikia** [comment]
Wait. Maybe its me who misunderstands. 
If the --force is responsible for changing the whole problem and not the event maybe thats not the right place to have it?
In my case only the config.yml file for the event changes and I need to kill the previous running event and deploy it again, but the problem itself is not changing. 
Maybe thats the wrong flag to use in this case?

**Comment by lucyleeow** [comment]
Yes, if by 'problem' the doc means the 'iris' challenge/problem, then we do not need `force` to re-deploy just one event.

**Comment by maikia** [comment]
If event is deployed, redeploying will not work without the force flag

**Comment by lucyleeow** [comment]
I think the problem is here:
https://github.com/paris-saclay-cds/ramp-board/blob/0fe2450f5cbc4f4c2204eb628631f4cf21ad775a/ramp-utils/ramp_utils/deploy.py#L79-L83

From the fun `deploy_ramp_event`:
```
force : bool, default is False
       Whether or not to potentially overwrite the repositories, problem and
       event in the database.
```
I don't know details but it seems that `force` can potentially overwrite the problem AND event, but we should keep this separate maybe?

cc @agramfort @rth

**Comment by lucyleeow** [comment]
I think the current/lazy 'solution' is to delete an event and make a new one, instead of overwritting an existing event

**Comment by agramfort** [comment]
just to be sure I follow. What do you expect from a redeploy?
all events use the same files (starting kit, problem etc). So when you
redeploy
it must affect all events right?

>

**Comment by maikia** [comment]
They use different config.yml

**Comment by lucyleeow** [comment]
re-deploy an event NOT a problem.
e.g., if we have an existing event, but need to change something in the event and re-deploy.
You can update simple things of an event but if you change significantly (e.g., we are altering the aws settings in the config.yml file of an event) I think you need to re-deploy

But if it isn't a common occurrence, we can stick with deleting event and re-deploying it.

**Comment by maikia** [comment]
Ok. Thats a solution but then we need to make it very clear. For now if you try to deploy running event it will fail suggesting using --force. There is no mention about other events being killed. There is no feedback on it. I just happened to notice that the others disappeared

**Comment by lucyleeow** [comment]
Yes there should definitely be a note, I'm just prioritising which bugs/enhancement to address

**Comment by agramfort** [comment]
when you redeploy what is the part you typically want to change?

**Comment by lucyleeow** [comment]
event `config.yml` file

**Comment by agramfort** [comment]
ok but more specifically the worker and the dispatcher?

**Comment by maikia** [comment]
the worker. I want to completely restart the worker with the new config.yml. That's why the solution suggested by @lucyleeow would work for my case.
"the current/lazy 'solution' is to delete an event and make a new one, instead of overwritting an existing event"

but this does not solve the problem of --force being confusing and risk of killing running events without users intention

**Comment by agramfort** [comment]
ok then it seems we have a quick solution and a more ambitious thing to fix
!

>

**Comment by lucyleeow** [comment]
Let's update the documentation for now. We can work on this later but I think the docker issue (#290) is higher priority.

**Comment by agramfort** [comment]
+1
